tasks/yard.rb

```diff
@@ -4,7 +4,9 @@
   require 'yard'
 
   YARD::Rake::YardocTask.new do |t|
-    t.options = ['-mmarkdown']
+    t.files = ['shoes-core/**/*.rb', 'README.md', 'LICENSE', 'CONTRIBUTING.md',
+               'CODE_OF_CONDUCT.MD']
+    t.options = ['-mmarkdown', '--no-private']
```

Comment:
So these are a good start, but I found generating locally that this gives me a bit better result:

```
  YARD::Rake::YardocTask.new do |t|
    t.files = [
      'shoes-core/lib/**/*.rb',
      '-',
      'README.md',
      'shoes-core/README.md',
      'LICENSE',
      'CONTRIBUTING.md',
      'CODE_OF_CONDUCT.MD'
    ]
    t.options = [
      '-mmarkdown',
      '--no-private'
    ]
```

It does a couple things:
* Only generates for the `shoes-core/lib` folder, since we were picking up specs
* Works correctly with the markdown files by putting them after a `-`

How does that look to you @nicholalexander?